### PR TITLE
Support google cloud for C++ chunk manager (#21102)

### PR DESCRIPTION
**Pull request by zhagnlu**
#21102

### CODE REVIEW

File: internal/core/src/storage/MinioChunkManager.h [hunk 1]
```diff
@@ -30,6 +42,9 @@
 
 namespace milvus::storage {
 
+enum RemoteStorageType { STORAGE_S3 = 0, STORAGE_GOOGLE_CLOUD = 1 };
```
Comment (xige-16) [inline internal/core/src/storage/MinioChunkManager.h:45]: 0 is not recommended as the valid enum value

File: internal/core/thirdparty/google_cloud_sdk/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    add_library(google_cloud_cpp_rest_internal SHARED IMPORTED)
+    set_target_properties(google_cloud_cpp_rest_internal
+        PROPERTIES
+        IMPORTED_GLOBAL TRUE
+        IMPORTED_LOCATION ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}google_cloud_cpp_rest_internal${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INTERFACE_INCLUDE_DIRECTORIES ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+    add_dependencies(google_cloud_cpp_rest_internal google_sdk_ep)
+endmacro()
+
+build_google_sdk_s3()
```
Comment (xige-16) [inline internal/core/thirdparty/google_cloud_sdk/CMakeLists.txt:142]: Could the imported gcp be compiled on mac？

File: internal/core/src/storage/MinioChunkManager.h (hunk 1 above)
Comment (czs007) [inline internal/core/src/storage/MinioChunkManager.h:45]: Prefer enum classes over “plain” enums

Comment (zhagnlu) [reply internal/core/src/storage/MinioChunkManager.h:45]: why?  0 is also default value of enum if you not set it

File: internal/core/thirdparty/google_cloud_sdk/CMakeLists.txt (hunk 2 above)
Comment (zhagnlu) [reply internal/core/thirdparty/google_cloud_sdk/CMakeLists.txt:142]: not valid it now, just support it at linux 
+# ******************************* Thirdparty google cloud sdk ********************************
+if ( LINUX AND BUILD_DISK_ANN STREQUAL "ON" )
+    add_subdirectory( google_cloud_sdk )
+endif()
+

File: internal/core/src/storage/MinioChunkManager.h (hunk 1 above)
Comment (zhagnlu) [reply internal/core/src/storage/MinioChunkManager.h]: update it

**Pull request merged**